src/components/Watchlater/Watchlater.jsx

```diff
@@ -1,7 +1,55 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import { useVideo } from "../../context";
+import { WatchlaterCard } from "../WatchlaterCard/WatchlaterCard";
+import "./watchlater.css";
 
 function Watchlater() {
-  return <div className="fx-grow">Watchlater</div>;
+  const navigate = useNavigate();
+  const [watchLaterVideos, setWatchLaterVideos] = useState([]);
+  const [fetchWatchlaterFlag, setFetchWatchlaterFlag] = useState(false);
+  const { setDisplayLoader } = useVideo();
+  const encodedToken = JSON.parse(localStorage.getItem("encodedToken"));
+  useEffect(() => {
+    if (encodedToken) {
+      setDisplayLoader(true);
+      (async () => {
+        try {
+          let res = await axios.get("/api/user/watchlater", {
+            headers: { authorization: encodedToken },
+          });
+          setWatchLaterVideos(res.data.watchlater);
```

Comment:
Adding a check whether the data you are receiving from your API is there or not, will prevent errors in case your API returns undefined.

Reply:
ok will add the check